benches/coprocessor_executors/util/bencher.rs

```diff
@@ -0,0 +1,108 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+use criterion::black_box;
+
+use tikv::coprocessor::dag::batch::interface::*;
+use tikv::coprocessor::dag::executor::Executor;
+use tikv::coprocessor::RequestHandler;
+
+pub trait Bencher {
+    fn bench(&mut self, b: &mut criterion::Bencher);
+}
+
+/// Invoke 1 next() of a normal executor.
+pub struct NormalNext1Bencher<E: Executor, F: FnMut() -> E> {
+    executor_builder: F,
+}
+
+impl<E: Executor, F: FnMut() -> E> NormalNext1Bencher<E, F> {
+    pub fn new(executor_builder: F) -> Self {
+        Self { executor_builder }
+    }
+}
+
+impl<E: Executor, F: FnMut() -> E> Bencher for NormalNext1Bencher<E, F> {
+    fn bench(&mut self, b: &mut criterion::Bencher) {
+        b.iter_batched_ref(
+            &mut self.executor_builder,
+            |executor| {
+                black_box(executor.next().unwrap());
+            },
+            criterion::BatchSize::SmallInput,
+        );
+    }
+}
+
+/// Invoke 1024 next() of a normal executor.
+pub struct NormalNext1024Bencher<E: Executor, F: FnMut() -> E> {
+    executor_builder: F,
+}
+
+impl<E: Executor, F: FnMut() -> E> NormalNext1024Bencher<E, F> {
+    pub fn new(executor_builder: F) -> Self {
+        Self { executor_builder }
+    }
+}
+
+impl<E: Executor, F: FnMut() -> E> Bencher for NormalNext1024Bencher<E, F> {
+    fn bench(&mut self, b: &mut criterion::Bencher) {
+        b.iter_batched_ref(
+            &mut self.executor_builder,
+            |executor| {
+                let iter_times = black_box(1024);
```

Comment:
The compiler may do some loop unrolling based on this constant loop times but in real world it should not happen (because even ourselves don't know how many times the loop should execute), so I wrapped it with a `black_box`.